app/src/main/java/com/duckduckgo/app/browser/DownloadConfirmationFragment.kt

```diff
@@ -24,46 +24,58 @@ import android.view.View
 import android.view.ViewGroup
 import android.widget.Toast
 import androidx.core.content.FileProvider.getUriForFile
-import com.duckduckgo.app.browser.downloader.NetworkFileDownloadManager.DownloadFileData
-import com.duckduckgo.app.browser.downloader.NetworkFileDownloadManager.UserDownloadAction
+import com.duckduckgo.app.browser.downloader.FileDownloader
+import com.duckduckgo.app.browser.downloader.guessFileName
 import com.duckduckgo.app.global.view.gone
 import com.duckduckgo.app.global.view.leftDrawable
 import com.duckduckgo.app.global.view.show
 import com.google.android.material.bottomsheet.BottomSheetDialogFragment
 import kotlinx.android.synthetic.main.download_confirmation.view.*
 import timber.log.Timber
+import java.io.File
+import java.io.IOException
+import kotlin.concurrent.thread
 
 class DownloadConfirmationFragment(
```

Comment:
Not related with this PR, but if we enable `Don't keep activities`, and we press home when this dialog is visible, and we bring the app from the background again, it crashes with:
`Unable to instantiate fragment com.duckduckgo.app.browser.DownloadConfirmationFragment: could not find Fragment constructor`

Reply:
Good catch! I've removed the custom constructor from the fragment so the init happens through the constructor and the bundle.